src/pspm_pfm.m

```diff
@@ -0,0 +1,588 @@
+            if any(structfun(@(x) x<1,lsts)), warning('ID:error_prepdata','An error occured in pspm_prepdata.'); return; end
+                
+            clear new_sr lsts
+        end
+    end
+    
+    % changing the sampling rate
+    sr = model.filter.down*ones(size(sr));
+    
+    % delete fields that are not useful for the model output
+    model.filter = rmfield(model.filter,'sr');
+    model.filter = rmfield(model.filter,'applied_filt');
+    filtered = 1;
+    
+% if data were not filtered
+else    
+    model = rmfield(model,'filter');
+    filtered = 0;
+end
+
+%%%%%%%% Determining mean values %%%%%%%%
+fprintf('Preparing for fitting ...\n')
+
+baseline_index = floor(sr(1)*model.baseline)+1;
+
+if exist('std_exp_cond','var')
+    tmp_data = [segm{:,std_exp_cond.ind}];
+    
+    std_exp_cond.data = nanmean([tmp_data.mean],2);
+    std_exp_cond.std = nanmean([tmp_data.std],2);
+    std_exp_cond.sem = nanmean([tmp_data.sem],2);
+end
+
+for i=1:nExpCond
+    
+    tmp_data = [segm{:,i}];
+    
+    tmp_data_new.data = nanmean([tmp_data.mean],2);
+    tmp_data_new.std = nanmean([tmp_data.std],2);
+    tmp_data_new.sem = nanmean([tmp_data.sem],2);
+    tmp_data_new.t = nanmean([tmp_data.t],2);
+    
+    % Subtracting the standard experimental condition
+    if exist('std_exp_cond','var') && i~=std_exp_cond.ind
+       tmp_data_new.data = tmp_data_new.data - std_exp_cond.data;
+       tmp_data_new.std = tmp_data_new.std + std_exp_cond.std;      % the error adds up
+       tmp_data_new.sem = tmp_data_new.sem + std_exp_cond.sem;      % the error adds up
+    end
+        
+    % Baselining data
+    tmp_data_new.data = tmp_data_new.data - tmp_data_new.data(baseline_index);
+    
+    % Dividing by the max value
+    if model.norm
+        [tmp_max,tmp_max_ind] = max(tmp_data_new.data);
+        tmp_data_new.data = tmp_data_new.data/tmp_max;
+        tmp_data_new.std = tmp_data_new.std + tmp_data_new.std(tmp_max_ind); % the error adds up
+        tmp_data_new.sem = tmp_data_new.sem + tmp_data_new.sem(tmp_max_ind); % the error adds up
+    end
+    
+    mean{1,i} = tmp_data_new;
+    
+    clear tmp_data tmp_data_new tmp_max tmp_max_ind
+end
+
+%%%%%%%% Fitting the model %%%%%%%%
+fprintf('Fitting ...\n')
+
+for i=1:nExpCond    
+    Y = mean{1,i}.data; 
+    
+    n = model.window;
+    td = n / length(Y);
+    model.bf.offset = 0.2;
+    
+    % Extending the size of the data vector in order to do the fitting,
+    % because if size(Y)=[n 1], the convolution would produce a vector of
+    % size [2*n-1 1] so we have to adjust Y accordingly. 
+    Y = [ Y ; zeros(size(Y,1)-1,size(Y,2))];
+    
+    % Creating an anonymous function
+    fun = @(x) norm(Y-conv(model.bf.fhandle([td;n;0;x]),model.bf.fhandle(td,n,model.bf.offset)).',2)^2;
+
+    % minimization
+    warning off all
+    [fitted{1,i}.optarg, fitted{1,i}.fval, sts] = fminsearch(fun,model.x0,options); 
+    warning on all
+    if sts == 0 
```

Comment:
Here, it might be a good idea to get the diagnostic output from `fminsearch` too using the inspectable call version (https://www.mathworks.com/help/matlab/ref/fminsearch.html#bvadxhn-11) in case the user wants to delve deeper into the possible optimization failures (or maybe make this optional)